unixfs/tar/writer.go

```diff
@@ -31,12 +32,18 @@ func DagArchive(ctx cxt.Context, nd *mdag.Node, name string, dag mdag.DAGService
 	// use a buffered writer to parallelize task
 	bufw := bufio.NewWriterSize(pipew, DefaultBufSize)
```

Comment:
Rather, `unixfs/io/dagreader.go` should be rewritten with `bufio` to get Flush automatically.